### PR TITLE
Add/date input component

### PR DESCRIPTION
**Pull request by alvarezmelissa87**
### Purpose
Related trello card: https://trello.com/c/LQEhw1rZ/133-reschedule-handler

Add DatePicker component.
Receives as props:
1. date to set current day at
2. a handler function for when the day selected changes.

Days in the past are disabled/not-selectable. 
### Things to Note
TODO: Add change in changelog after merge when updating version.
TODO: Add test coverage for failing coverage check on this component.
### Review
@hharnisc - would love to hear your thoughts on the direction herel and chat on adding more coverage. Thank you! 😸 
Also - curious how in-depth we want to go with the date calculation? Right now we allow selecting the day if it is the day of - perhaps we want to be more accurate and only allow selecting day of if there are 3 hours left in the day or something. Would love to chat around this a bit more. 
cc @pioul for context. 😄 
**Some aspects to consider during review:**
* Functionality and implementation
* Architecture and performance
* Code readability and style

_A friendly reminder to add a reviewer/reviewers, add an assignee (yourself if you've worked on the change), set the code review status, and set the component request. :smile:_

### CODE REVIEW

File: DateInput/index.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const DateInput = ({
+  onChange,
+  scheduledAt,
+}) =>
+  <span>
+    <DayPicker
+      disabledDays={inThePast}
+      fromMonth={new Date()}
+      onDayClick={ (day, { disabled }) => {
+        if (disabled) return;
```
Comment (alvarezmelissa87) [inline DateInput/index.jsx]: If day is not marked as `disabled` allow `onChange` handler to be executed. 
`onChange` handler is passed in as prop -  will handle the updating of the store with the new selected day. 
Store change triggers re-render, which allows new selected day value to be passed in to `DateInput` as `scheduledAt` - causing newly selected day to be highlighted.

File: DateInput/style.css [hunk 2]
```diff
@@ -0,0 +1,161 @@
+/**
```
Comment (alvarezmelissa87) [inline DateInput/style.css]: Style consistent with Multiple Composers.

File: DateInput/index.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const inThePast = (day) => {
+  const now = new Date();
+  const isToday = day.toDateString() === now.toDateString();
+  return (day < now) && !isToday;
+};
+
+const DateInput = ({
+  onChange,
+  value,
+}) =>
```
Comment (alvarezmelissa87) [inline DateInput/index.jsx]: @hharnisc - updated the terminology  here to 'value' to fit what redux-form will be expecting. I _think_ that was the only thing that we needed for this. Curious if this feels ready to go? Thanks! 😄

**Comment by hharnisc** [comment]
pending this next commit passing CI - I think this this ready to go @alvarezmelissa87 @stevemdixon

**Comment by hharnisc** [comment]
Gonna squash and merge this into master!

**Pull request merged**